fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/service/SavingsSchedularServiceImpl.java

```diff
@@ -65,23 +86,61 @@ public void postInterestForAccounts() throws JobExecutionException {
-                    LocalDate transactionDate = null;
-                    this.savingsAccountWritePlatformService.postInterest(savingsAccount, postInterestAsOn, transactionDate);
-                } catch (Exception e) {
-                    LOG.error("Failed to post interest for Savings with id {}", savingsAccount.getId(), e);
-                    errors.add(e);
-                }
+
+        if (this.globalConfigurationRepository.findOneByName("allow-split-interest-posting-on-savings").isEnabled()) {
+            final ScheduledJobDetail scheduledJobDetail = this.schedularWritePlatformService.retrieveJobByNameAndNodeId(this.nodeId,
+                    JobName.POST_INTEREST_FOR_SAVINGS.toString());
+            final String officeId = this.jobParameterRepository
+                    .findByJobParameterByJobIdAndParameterName(scheduledJobDetail.getId(), "officeId").getParameterValue();
+
+            if (officeId == null) {
+                throw new OfficeIdParameterNotFound(scheduledJobDetail.getId());
             }
-            page++;
-            totalPageSize = savingsAccounts.getTotalPages();
-        } while (page < totalPageSize);
+
+            List<Office> offices = this.officeReadPlatformService.retrieveAllOfficesForInterestPostingOnNode(Long.getLong(officeId));
+            for (Office office : offices) {
+                // Load clients data
+                office.loadClients();
+                do {
+                    PageRequest pageRequest = PageRequest.of(page, initialSize);
+                    Page<SavingsAccount> savingsAccounts = this.savingsAccountRepository.findByStatus(ACTIVE.getValue(), pageRequest);
+                    for (SavingsAccount savingsAccount : savingsAccounts.getContent()) {
+                        if (!office.getClients().contains(savingsAccount.getClient())) {
+                            continue;
+                        }
+                        try {
+                            this.savingAccountAssembler.assignSavingAccountHelpers(savingsAccount);
+                            boolean postInterestAsOn = false;
+                            LocalDate transactionDate = null;
+                            this.savingsAccountWritePlatformService.postInterest(savingsAccount, postInterestAsOn, transactionDate);
+                        } catch (Exception e) {
+                            LOG.error("Failed to post interest for Savings with id {}", savingsAccount.getId(), e);
+                            errors.add(e);
+                        }
+                    }
+                    page++;
+                    totalPageSize = savingsAccounts.getTotalPages();
+                } while (page < totalPageSize);
+            }
+        } else {
+            do {
+                PageRequest pageRequest = PageRequest.of(page, initialSize);
+                Page<SavingsAccount> savingsAccounts = this.savingsAccountRepository.findByStatus(ACTIVE.getValue(), pageRequest);
```

Comment:
this if..else repeats same code. instead of repeating the code ( which will be difficult to maintain) move the repeating code to a method and reuse.